libraries/abstractions/common_io/test/test_iot_internal.h

```diff
@@ -33,6 +33,190 @@
+ #define IOT_TEST_COMMON_IO_GPIO_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_RTC_SUPPORTED
+ #define IOT_TEST_COMMON_IO_RTC_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_TIMER_SUPPORTED
+ #define IOT_TEST_COMMON_IO_TIMER_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_ADC_SUPPORTED
+ #define IOT_TEST_COMMON_IO_ADC_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_RESET_SUPPORTED
+ #define IOT_TEST_COMMON_IO_RESET_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_PERFCOUNTER_SUPPORTED
+ #define IOT_TEST_COMMON_IO_PERFCOUNTER_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_PWM_SUPPORTED
+ #define IOT_TEST_COMMON_IO_PWM_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_I2C_SUPPORTED
+ #define IOT_TEST_COMMON_IO_I2C_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_TEMP_SENSOR_SUPPORTED
+ #define IOT_TEST_COMMON_IO_TEMP_SENSOR_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_POWER_SUPPORTED
+ #define IOT_TEST_COMMON_IO_POWER_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_BATTERY_SUPPORTED
+ #define IOT_TEST_COMMON_IO_BATTERY_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_EFUSE_SUPPORTED
+ #define IOT_TEST_COMMON_IO_EFUSE_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_SPI_SUPPORTED
+ #define IOT_TEST_COMMON_IO_SPI_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_USB_DEVICE_SUPPORTED
+ #define IOT_TEST_COMMON_IO_USB_DEVICE_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_USB_HOST_SUPPORTED
+ #define IOT_TEST_COMMON_IO_USB_HOST_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_SDIO_SUPPORTED
+ #define IOT_TEST_COMMON_IO_SDIO_SUPPORTED 1
+#endif
+
+#ifndef IOT_TEST_COMMON_IO_I2S_SUPPORTED
+ #define IOT_TEST_COMMON_IO_I2S_SUPPORTED 1
+#endif
+
+
+/*
+ * The variables below are used in the tests. Generally. they should be
+ * set in SET_TEST_IOT_RESET_CONFIG() defined in the respective platform
+ * directory. If not set, the default value set at the variable
+ * definition is used.
+ */
+
+/* Uart */
+extern uint8_t ustestIotUartPort;                       /* the index of the UART that will be tested */
+
+/* Flash */
+extern uint32_t ultestIotFlashStartOffset;              /* the Flash offset at which the flash operations in the test will take place */
+
+/* GPIO */
+extern int32_t ltestIotGpioPortA;                       /* The 1st GPIO port used in the loop back test */
+extern int32_t ltestIotGpioPortB;                       /* The 2nd GPIO port used in the loop back test */
+extern uint16_t ustestIotGpioConfig;                    /* ustestIotGpioConfig is used for configuring GPIO test
+                                                         * user must set this variable for testing different GPIOs
+                                                         *
+                                                         * default ustestIotGpioConfig = 0 means reading logic GPIO# 0
+                                                         *
+                                                         * bit[7:0] for port
```

Comment:
Are these bit definitions defined anywhere?  If so, just refer to the symbols here rather than defining them in comments.

Reply:
They are defined as macros in test_iot_gpio.c
Here is how they look in that file, for your reference:

```
/*-----------------------------------------------------------*/
#define testIotGPIO_TEST_PORT_MASK  ( 0x00FF )
#define testIotGPIO_TEST_DIR_BIT    ( 8 )
#define testIotGPIO_TEST_DIR_MASK   ( 1 << testIotGPIO_TEST_DIR_BIT )
#define testIotGPIO_TEST_IRQ_BIT    ( 9 )
#define testIotGPIO_TEST_IRQ_MASK   ( 1 << testIotGPIO_TEST_IRQ_BIT )
#define testIotGPIO_TEST_VALUE_BIT  ( 10 )
#define testIotGPIO_TEST_VALUE_MASK ( 1 << testIotGPIO_TEST_VALUE_BIT )
#define testIotGPIO_TEST_READ_DELAY ( 50 )
/*-----------------------------------------------------------*/
```
All the macros above except the last one are mapped to this bits in the comment under review.

Reply:
Can you add a reference to this file and/or these macros in the comments here?  The point is to make it so that someone reading this code understands where these definitions actually exist; this will help them from getting out of sync.

Reply:
Done.